examples/complex-tests.dx

```diff
@@ -0,0 +1,54 @@
+:p complex_floor $ MkComplex 0.3 0.6
+> (MkComplex 0.0 0.0)
+:p complex_floor $ MkComplex 0.6 0.8
+> (MkComplex 0.0 1.0)
+:p complex_floor $ MkComplex 0.8 0.6
+> (MkComplex 1.0 0.0)
+:p complex_floor $ MkComplex 0.6 0.3
+> (MkComplex 0.0 0.0)
+
+a = MkComplex 2.1 0.4
+b = MkComplex (-1.1) 1.3
+:p (a + b - a) ~~ b
+> True
+:p (a * b) ~~ (b * a)
+> True
+:p fdiv (a * b) a ~~ b
+> True
+--:p ((a * b) / a) ~~ b
+--> True
+:p a == b
+> False
+:p a == a
+> True
+:p log (exp a) ~~ a
+> True
+:p exp (log a) ~~ a
+> True
+:p log2 (exp2 a) ~~ a
+> True
+:p exp2 (log2 a) ~~ a
+> True
+:p sqrt (sq a) ~~ a
+> True
+:p log ((MkComplex 1.0 0.0) + a) ~~ log1p a
+> True
+:p sin (-a) ~~ (-(sin a))
+> True
+:p cos (-a) ~~ cos a
+> True
+:p tan (-a) ~~ (- (tan a))
+> True
+:p exp (pi .* (MkComplex 0.0 1.0)) ~~ (MkComplex (-1.0) 0.0)  -- Euler's identity
+> True
+:p ((sq (sin a)) + (sq (cos a))) ~~ (MkComplex 1.0 0.0)
+> True
+:p complex_abs b > 0.0
+> True
+
+:p complex_sinh (MkComplex 1.2 3.2)
+> (MkComplex -1.5068874 -0.10569556)
+:p complex_cosh (MkComplex 1.2 3.2)
```

Comment:
Done, although I couldn't think of a better name, and I think they should probably be part of the `Floating` class.  If I convert `Floating` to the new `interface` syntax I might move them there, if you think that'd make sense.

Reply:
oh yeah I meant `Floating` of course! Yes, converting more type classes to use the `interface` syntax would be super helpful!

Reply:
Okay, that makes more sense!  I thought maybe you had some intuition for the hidden unity of division and `sinh` that I was unaware of.  I'll move those to `Floating`.

One question in my mind is:  Do we want to move everything to `interface` and deprecate the `data` keyword?  For now I'm just converting the interfaces that have functions in them, but not e.g. `data Bool`.

Reply:
Haha no, just a typo. Feel free to call out things that don't seem to make sense 😛 

`data` will stay around, but only for things that actually do contain data. E.g. `Bool` is a legit value that you might want to do some compute with, but all `Floating` instances exist purely for the purpose of providing some polymorphic functionality. Also, you do often want to directly refer to the constructors of `Bool` (i.e. `True` and `False)`, while you generally don't care about the constructor of an interface. Internally the `interface`s are implemented using `data`, but they often do have a different meaning semantically.

A rule of thumb I would use is: if something always comes before the `?=>` arrow, then it's an `interface`. Otherwise it is `data`.

Reply:
Thanks for explaining, that makes sense to me.